lib/MongoDB/_URI.pm

```diff
@@ -442,6 +441,14 @@ sub _parse_srv_uri {
       %{ $result{options} || {} },
     };
 
+    # Reset str to bool options to string value, as _parse_options changes it to 0/1 if it exists during parsing
+    # means we get the correct value when re-building the uri below.
```

Comment:
Seems right, but can we get some tests for it?

Reply:
Actually there are tests for this in devel, from ticket PERL-1061 - whcih means this was a regression when URI got re-factored...

Reply:
Figured out a way to test it without needing DNS!